### ISSUE TITLE
rspc can only handle one request at a time

### ISSUE THREAD

**Issue by campbellcole** [oscartbeaumont/rspc#260]
I don't know if this is considered a bug but I wanted to get clarification on if this is intended. I have a RSPC router that uses a mixture of sync and async procedures. I am trying to implement a system that allows certain procedure calls to require confirmation. The way I've implemented this is by blocking in the procedure body until another procedure call unblocks it.

The issue is that the second procedure call never makes it through because the first one is still waiting. This results in a deadlock.

To me, it doesn't make sense why RSPC would only be able to handle requests sequentially given that RSPC allows async procedures.

I could work around this by spawning another thread/task in the IPC call so it returns immediately, but that would mean I'd lose the return value of the procedure were it to fail. Is there any way to easily lift this restriction, and is this even intended?

Edit: this is in a `tauri` app btw. It seems like the issue may be in the way the plugin is set up, where it spawns an async task to `rx.recv` but `await`s each request instead of spawning a separate task?

**Comment by campbellcole** [comment]
Ok I am happy to say that I have discovered the source of this and found a trivial fix. I'll submit a pull request shortly. This issue likely only affects the Tauri plugin.

**Comment by oscartbeaumont** [comment]
Okay yeah that will do it. This has been fixed on `main` for a long time so I completely forgot it was a possibility.

Will push this fix and a couple other things as `0.1.4`.
